### PR TITLE
nvme-print: Fix to add uint and uint64 json object values to output

### PR DESCRIPTION
**Pull request by ikegami-t**
This fix is for the issue #1742 opened.

Signed-off-by: Tokunori Ikegami <ikegami.t@gmail.com>

### CODE REVIEW

File: nvme-print.c [hunk 1]
```diff
@@ -169,13 +169,13 @@ static void json_nvme_id_ns(struct nvme_id_ns *ns, bool cap_only)
 		}
 
 		json_object_add_value_int(root, "mssrl", le16_to_cpu(ns->mssrl));
-		json_object_add_value_int(root, "mcl", le32_to_cpu(ns->mcl));
+		json_object_add_value_uint(root, "mcl", le32_to_cpu(ns->mcl));
```
Comment (igaw) [inline nvme-print.c:172]: We should also update the normal terminal output for this value. The printf format string `%d` is used currently.

File: nvme-print.c [hunk 2]
```diff
@@ -281,19 +281,19 @@ static void json_nvme_id_ctrl(struct nvme_id_ctrl *ctrl,
 	json_object_add_value_int(root, "hctma", le16_to_cpu(ctrl->hctma));
 	json_object_add_value_int(root, "mntmt", le16_to_cpu(ctrl->mntmt));
 	json_object_add_value_int(root, "mxtmt", le16_to_cpu(ctrl->mxtmt));
-	json_object_add_value_int(root, "sanicap", le32_to_cpu(ctrl->sanicap));
-	json_object_add_value_int(root, "hmminds", le32_to_cpu(ctrl->hmminds));
+	json_object_add_value_uint(root, "sanicap", le32_to_cpu(ctrl->sanicap));
+	json_object_add_value_uint(root, "hmminds", le32_to_cpu(ctrl->hmminds));
```
Comment (igaw) [inline nvme-print.c:285]: also for `hmminds` we should update the `%d` format specifier.

Could you double check that all the print statements for JSON and terminal output agree on the output format? Thanks!

Comment (ikegami-t) [reply nvme-print.c:285]: Just pushed the additional patch to update the mcl and hmminds normal terminal output print format.
Also just checked the outputs as [test.txt](https://github.com/linux-nvme/nvme-cli/files/10260841/test.txt) but sorry some command outputs are not supported on my environment so not tested actually.

**Comment by igaw** [comment]
FWIW, I've updated all places where we print a uint32_t type to use `%u` or 
`json_object_add_value_uint()`

One thing I of 'beauty' I stumbled upon is

```c
        long long lba = nvme_ns_get_lba_size(n);
        double nsze = nvme_ns_get_lba_count(n) * lba;
        double nuse = nvme_ns_get_lba_util(n) * lba;

        const char *s_suffix = suffix_si_get(&nsze);
        const char *u_suffix = suffix_si_get(&nuse);
        const char *l_suffix = suffix_binary_get(&lba);
```
`nvme_ns_get_lba_size()` returns an `int`, `nvme_ns_get_lba_count()` return a `uint64_t` and `suffix_si_get()` does internally do a float type division. IIRC, this is not going to work correctly on 32 bit platforms.

**Comment by igaw** [comment]
Fixes: #1742

**Comment by ikegami-t** [comment]
> FWIW, I've updated all places where we print a uint32_t type to use `%u` or `json_object_add_value_uint()`
> 
> One thing I of 'beauty' I stumbled upon is
> 
> ```c
>         long long lba = nvme_ns_get_lba_size(n);
>         double nsze = nvme_ns_get_lba_count(n) * lba;
>         double nuse = nvme_ns_get_lba_util(n) * lba;
> 
>         const char *s_suffix = suffix_si_get(&nsze);
>         const char *u_suffix = suffix_si_get(&nuse);
>         const char *l_suffix = suffix_binary_get(&lba);
> ```
> 
> `nvme_ns_get_lba_size()` returns an `int`, `nvme_ns_get_lba_count()` return a `uint64_t` and `suffix_si_get()` does internally do a float type division. IIRC, this is not going to work correctly on 32 bit platforms.

Thanks for your reviewing and commnets.
Just FYI: About the 32 bit platforms issue mentioned by you I have just tested the code on my local 32 bit platform envinronment then seems it works correctly as same with the 64 bit plotform as same output printed so looks no problem.

**Comment by igaw** [comment]
Thanks for testing. I got properly confused with all the type conversion.

**Pull request merged**